### PR TITLE
Generate nodeid from local file

### PR DESCRIPTION
**Pull request by felipemadero**
- print nodeid on node status and node list, generating it from local file
- generate nodeid from local file for node validate
- remove obsolete script to get node id from remote calls
- encapsulate mapping ansible node id <-> node id, cloud service

Closes https://github.com/ava-labs/avalanche-cli/issues/1105 https://github.com/ava-labs/avalanche-cli/issues/1143 https://github.com/ava-labs/avalanche-cli/issues/1142 https://github.com/ava-labs/avalanche-cli/issues/1137

### CODE REVIEW

File: pkg/ansible/ansible.go [hunk 1]
```diff
@@ -439,3 +424,29 @@ func UpdateInventoryHostPublicIP(inventoryDirPath string, nodesWoEIP map[string]
 	}
 	return nil
 }
+
+func ToAnsibleInstanceID(cloudService string, instanceID string) (string, error) {
```
Comment (arturrez) [inline pkg/ansible/ansible.go:428]: should be moved to models.Host

File: pkg/ansible/ansible.go [hunk 2]
```diff
@@ -439,3 +424,29 @@ func UpdateInventoryHostPublicIP(inventoryDirPath string, nodesWoEIP map[string]
...
+	switch cloudService {
+	case constants.GCPCloudService:
+		return fmt.Sprintf("%s_%s", constants.GCPNodeAnsiblePrefix, instanceID), nil
+	case constants.AWSCloudService:
+		return fmt.Sprintf("%s_%s", constants.AWSNodeAnsiblePrefix, instanceID), nil
+	}
+	return "", fmt.Errorf("unknown cloud service %s", cloudService)
+}
+
+func FromAnsibleInstanceID(ansibleInstanceID string) (string, string, error) {
```
Comment (arturrez) [inline pkg/ansible/ansible.go:438]: should be moved to models.Hosts

File: pkg/ansible/ansible.go [hunk 3]
```diff
@@ -439,3 +424,29 @@ func UpdateInventoryHostPublicIP(inventoryDirPath string, nodesWoEIP map[string]
...
+	splitID := strings.Split(ansibleInstanceID, "_")
+	if len(splitID) < 2 {
+		return "", "", fmt.Errorf("invalid format on ansible instance id %s", ansibleInstanceID)
+	}
+	cloudServicePrefix := strings.Join(splitID[:len(splitID)-1], "_")
```
Comment (arturrez) [inline pkg/ansible/ansible.go:443]: maybe use something like ? https://github.com/ava-labs/avalanche-cli/pull/1121/files#diff-4ec42bf2e112e7f2bc4260ba7cde4a78513dd9ffa3da7d7b3efc6aba315eeb32R39

File: pkg/utils/common.go [hunk 4]
```diff
@@ -64,3 +64,23 @@ func GetRealFilePath(path string) string {
 	}
 	return path
 }
+
+func Map[T, U any](input []T, f func(T) U) []U {
```
Comment (arturrez) [inline pkg/utils/common.go:68]: is it used?

**Review by arturrez — APPROVED** [review]

File: pkg/utils/common.go (hunk 4 above)
Comment (felipemadero) [reply pkg/utils/common.go:78]: added a case

**Pull request merged**